run/django/e2e_test.py

```diff
@@ -0,0 +1,406 @@
+# Copyright 2020 Google, LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This test creates a Cloud SQL instance, a Cloud Storage bucket, associated
+# secrets, and deploys a Django service
+
+import os
+import subprocess
+import uuid
+
+from google.cloud import secretmanager_v1 as sm
+import pytest
+import requests
+
+# Unique suffix to create distinct service names
+SUFFIX = uuid.uuid4().hex[:10]
+
+PROJECT = os.environ["GOOGLE_CLOUD_PROJECT"]
+REGION = "us-central1"
+POSTGRES_INSTANCE = os.environ["POSTGRES_INSTANCE"]
+
+# Most commands in this test require the short instance form
+if ":" in POSTGRES_INSTANCE:
+    POSTGRES_INSTANCE = POSTGRES_INSTANCE.split(":")[-1]
+
+CLOUD_STORAGE_BUCKET = f"{PROJECT}-media-{SUFFIX}"
+
+POSTGRES_DATABASE = f"polls-{SUFFIX}"
+POSTGRES_USER = f"django-{SUFFIX}"
+POSTGRES_PASSWORD = uuid.uuid4().hex[:26]
+
+ADMIN_NAME = "admin"
+ADMIN_PASSWORD = uuid.uuid4().hex[:26]
+
+SECRET_SETTINGS_NAME = f"django_settings-{SUFFIX}"
+SECRET_PASSWORD_NAME = f"superuser_password-{SUFFIX}"
+
+
+@pytest.fixture
+def project_number():
+    projectnum = (
+        subprocess.run(
+            [
+                "gcloud",
+                "projects",
+                "list",
+                "--filter",
+                f"name={PROJECT}",
+                "--format",
+                "value(projectNumber)",
+            ],
+            stdout=subprocess.PIPE,
+            check=True,
+        )
+        .stdout.strip()
+        .decode()
+    )
+    print("DEBUG:", projectnum)
+    yield projectnum
+
+
+@pytest.fixture
+def postgres_host(project_number):
```

Comment:
@kurtisvg is postgres one of the few static instances that we actually are okay with not having a fixture for because it takes super long to spin up and tear down?

Reply:
Yes (see the [Sample Style Guide](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md#external-resources)).

The instance should be provided via env var, however either the database or the table(s) used should be created on the fly (and should use a UUID).

Reply:
Would you like inline documentation about this choice, or can I resolve this conversation without any edits?